yt_dlp/extractor/downdogapp.py

```diff
@@ -0,0 +1,191 @@
+
+        # Build the POST data string, order matters
+        timings_data = (
+            f'sequenceId={sequence_id}&'
+            f'playlistSettings=%7B%2216%22%3A-1%7D&'
+            f'forOffline=false&'
+            f'deviceDescription={device}&'
+            f'osVersion={os}&'
+            f'timeZone={timezone}&'
+            f'locale=en-US&'
+            f'appType=ORIGINAL&'
+            f'timestamp={unix_timestamp}&'
+            f'cred={api_key}&'
+            f'languageOption={language}'
+        ).encode()
+
+        timings = self._request_webpage(
+            f'https://{hostname}/json/practice',
+            sequence_id,
+            data=timings_data,
+            headers={'Content-Type': 'application/x-www-form-urlencoded'},
+        )
+        timings_dict = self._parse_json(timings.read().decode('utf-8'), sequence_id)
+        try:
+            length_in_minutes = int(
+                timings_dict['sequence']['totalTime']['seconds'] // 60,
+            )
+            title = (
+                f'Down Dog {timings_dict["sequence"]["activityType"].capitalize()} -'
+                f' {timings_dict["sequence"]["trackerLabel"]} {length_in_minutes}min'
+            )
+        except KeyError:
+            length_in_minutes = 80  # to be safe
+            title = None
+
+        playback_post_data = (
+            f'sequenceId={sequence_id}&'
+            f'videoOffsetTime=0&'
+            f'audioBalance=0.5&'
+            f'includeCountdown=true&'
+            f'includeOverlay=true&'
+            f'includePoseNames=true&'
+            f'includeClosedCaptions=false&'
+            f'mirrorVideo=false&'
+            f'videoQualityId=4&'
+            f'cellularConnection=false&'
+            f'chromecast=false&'
+            f'airplay=false&'
+            f'deviceDescription={device}&'
+            f'osVersion={os}&'
+            f'timeZone={timezone}&'
+            f'locale=en-US&'
+            f'appType=ORIGINAL&'
+            f'timestamp={unix_timestamp}&'
+            f'cred={api_key}&'
+            f'languageOption={language}'
+        ).encode()
+
+        response = self._request_webpage(f'https://{hostname}/json/playbackUrl',
+                                         sequence_id,
+                                         data=playback_post_data,
+                                         headers={'Content-Type': 'application/x-www-form-urlencoded'},
+                                         )
+        response_data = response.read().decode('utf-8')
+        api_data = self._parse_json(response_data, sequence_id)
+        m3u8_url = api_data['url']
+
+        # wait for video to be 'stitched' by the server before downloading,
+        #  otherwise will only get a partial file
+        sleep_minutes = length_in_minutes / 6
+        self.report_warning(
+            f'Waiting for media to be created by the server, starting download in {sleep_minutes:.1f} minutes...',
+        )
+        time.sleep(sleep_minutes * 60)
```

Comment:
this is obviously really undesirable.

Reply:
Very strange tbh. How does the app handle this?

If it's necessary, then it's necessary. however:
1. use `self._sleep` instead of `time.sleep`
2. don't bother converting to minutes above
3. sleeping should only happen `if not self.get_param('simulate')`